src/pyramid/security.py

```diff
@@ -244,6 +244,11 @@ def authenticated_userid(self):
             return None
         return policy.authenticated_userid(self)
 
+    @property
+    def is_authenticated(self):
+        """Return True if a user is authenticated for this request."""
+        return self.authenticated_identity is not None
```

Comment:
To do: confirm that this is the right attribute to check.  I have forgotten if identity/userid depend on each other, and if yes which one is the primary one.

Reply:
this is ok for now